src/database.ts

```diff
@@ -310,7 +310,41 @@ class Database extends common.GrpcServiceObject {
         options: CreateDatabaseOptions,
         callback: CreateDatabaseCallback
       ) => {
-        return instance.createDatabase(formattedName_, options, callback);
+        const pool = this.pool_ as SessionPool;
+        if (pool._pending > 0) {
+          // If there are BatchCreateSessions requests pending, then we should
+          // wait until these have finished before we try to create the database.
+          // Otherwise the results of these requests might be propagated to
+          // client requests that are submitted after the database has been
+          // created. If the pending requests have not finished within 10 seconds,
+          // they will be ignored and the database creation will proceed.
+          let timeout;
+          const promises = [
+            new Promise<void>(
+              resolve => (timeout = setTimeout(resolve, 10000))
+            ),
+            new Promise<void>(resolve => {
+              pool
+                .on('available', () => {
+                  if (pool._pending === 0) {
+                    clearTimeout(timeout);
+                    resolve();
+                  }
+                })
+                .on('createError', () => {
+                  if (pool._pending === 0) {
+                    clearTimeout(timeout);
+                    resolve();
+                  }
+                });
+            }),
+          ];
+          Promise.race(promises).then(() =>
+            instance.createDatabase(formattedName_, options, callback)
```

Comment:
At this point, I guess the fill() call has failed because we haven't created the database yet. Then we create the database. When will the fill() call be retried?

I just want to make sure that we don't get into a situation where the database is created but the session pool is never initialized.

Reply:
Good point, that never happened. The pool would be gradually filled as the user application is requesting sessions, and if the user application was using the default session pool settings this would also happen in one `BatchCreateSessions` RPC (because of the default `incStep: 25, min: 25`. But with custom settings such as for example:

```typescript
min: 25,
max: 400,
incStep: 10,
```

The minimum configured value would not be created directly.

I've added that logic to the `_getSession` function where the dynamic session creation happens, so it also works for any other scenario where the initial filling of the session pool failed.